src/python/zquantum/qaoa/estimators.py

```diff
@@ -45,35 +44,49 @@ def get_estimated_expectation_values(
         Returns:
             ExpectationValues: expectation values for each term in the target operator.
         """
-        if alpha > 1 or alpha <= 0:
+        if self.alpha > 1 or self.alpha <= 0:
             raise ValueError("alpha needs to be a value between 0 and 1.")
 
-        if not isinstance(target_operator, IsingOperator):
-            raise TypeError("Operator should be of type IsingOperator.")
-
-        distribution = backend.get_bitstring_distribution(circuit, n_samples=n_samples)
-        expected_values_per_bitstring = {}
+        circuits, operators, shots_per_circuit = zip(
+            *[(e.circuit, e.operator, e.number_of_shots) for e in estimation_tasks]
+        )
+        distributions_list = [
+            backend.get_bitstring_distribution(circuit, n_samples=n_shots)
+            for circuit, n_shots in zip(circuits, shots_per_circuit)
+        ]
 
-        for bitstring in distribution.distribution_dict:
-            expected_value = Measurements(bitstring).get_expectation_values(
-                target_operator
+        return [
+            ExpectationValues(
+                np.array(
+                    _calculate_expectation_value_for_distribution(
+                        distribution, operator, self.alpha
+                    )
+                )
             )
-            expected_values_per_bitstring[bitstring] = expected_value.values[0]
+            for distribution, operator in zip(distributions_list, operators)
+        ]
 
-        sorted_expected_values_per_bitstring_list = sorted(
-            expected_values_per_bitstring.items(), key=lambda item: item[1]
-        )
 
-        cumulative_prob = 0.0
-        cumulative_value = 0.0
+def _calculate_expectation_value_for_distribution(distribution, operator, alpha):
```

Comment:
A docstring and/or more comments throughout this function would help a lot for future readers. Especially since this is really the "core" of CVaR, I think it would be good to describe more what is happing with inline comments.

Reply:
I didn't want to overdo it, so I added just inline comments, as it's well described in the class description.
See: https://github.com/zapatacomputing/z-quantum-qaoa/pull/44/commits/58acb9946120ecbe079898018c01eec06cc6d596